cmake/compiler/gcc/target_security_fortify.cmake

```diff
@@ -5,9 +5,13 @@ macro(toolchain_cc_security_fortify)
     # _FORTIFY_SOURCE: Detect common-case buffer overflows for certain functions
     # _FORTIFY_SOURCE=1 : Compile-time checks (requires -O1 at least)
     # _FORTIFY_SOURCE=2 : Additional lightweight run-time checks
-    zephyr_compile_definitions(
-      _FORTIFY_SOURCE=2
-    )
+
+    if(NOT CMAKE_C_COMPILER_ID STREQUAL "Clang")
+      # clang errors on _FORTIFY_SOURCE=2
+      zephyr_compile_definitions(
+        _FORTIFY_SOURCE=2
+	)
+    endif()
```

Comment:
This file is for gcc, but until now clang has been reusing gcc stuff. But now there is a difference, so now clang does need its own `target_security_fortify.cmake`.